packages/trace-viewer/src/sw/main.ts

```diff
@@ -93,12 +93,20 @@ async function doFetch(event: FetchEvent): Promise<Response> {
       return new Response(null, { status: 200 });
     }
 
+    let traceViewerServerBaseUrl = self.registration.scope;
+    if (client?.url) {
+      const clientUrl = new URL(client.url);
+      if (clientUrl.searchParams.has('server'))
+        traceViewerServerBaseUrl = clientUrl.searchParams.get('server')!;
+    }
+    const traceViewerServer = new TraceViewerServerBackend(traceViewerServerBaseUrl);
```

Comment:
It feels wrong to create this object for each `fetch()`, even though it's stateless. Can we store it next to `client` instead?

Reply:
You mean like in https://github.com/microsoft/playwright/pull/33542/commits/592f7cc2e33ec145c6e3ee4a72a740562b6f2870?